crates/types/src/bound.rs

```diff
@@ -0,0 +1,78 @@
+use {
+    crate::{StdError, StdResult},
+    std::{marker::PhantomData, ops::Bound},
```

Comment:
What I'm not sure about this implementation is it uses `std::ops::Bound`, while we have another `grug_storage::Bound` of the same name. This can be confusing.

Should we move `grug_storage::Bound` here and use it instead?

Reply:
We're now using `grug_types::Bound`.